Mention the server-version for migrations (#398)
doctrine/DoctrineMigrationsBundle: Resources/doc/index.rst

Note that if you have multiple connections configured then the ``schema_filter`` configuration
will need to be placed per-connection.

> Migrations Metadata Storage

Reviewer: This is more a troubleshooting than a chapter for the metadata storage. Maybe you can rewrite it to make it sound more like a troubleshooting.

Reviewer 2: ```suggestion
Troubleshooting out of sync metadata storage issue
```

> ---------------------------

Reviewer 2: ```suggestion
--------------------------------------------------
```

> DoctrineMigrations rely on a properly configured Database server version in the connection string to manage the table storing the

Reviewer 2: ```suggestion
``doctrine/migrations`` relies on a properly configured Database server version in the connection string to manage the table storing the
```

> migrations, also known as the MetadataStorage.

Reviewer 2: ```suggestion
migrations, also known as the metadata storage.
```

> If you encounter the error `The metadata storage is not up to date, please run the sync-metadata-storage command to fix this issue.`

Reviewer 2: ```suggestion
If you encounter the error ``The metadata storage is not up to date, please run the sync-metadata-storage command to fix this issue.``
```

> when running the command `doctrine:migrations:migrate` or the suggested command itself `doctrine:migrations:sync-metadata-storage` please

Reviewer 2: ```suggestion
when running the command ``doctrine:migrations:migrate`` or the suggested command itself ``doctrine:migrations:sync-metadata-storage`` please
```

check the database connection string, and make sure that the proper server version is defined. If you are running a MariaDB database,

> you should prefix the server version with `mariadb-` (ex: `mariadb-10.2.12`). See the `configuring_database`_ section.

Reviewer 2: ```suggestion
you should prefix the server version with ``mariadb-`` (ex: ``mariadb-10.2.12``). See the `configuring_database`_ section.
```

Example connection string for MariaDB:

.. code-block:: terminal
    DATABASE_URL=mysql://root:@127.0.0.1:3306/testtest?serverVersion=mariadb-10.4.11

> 

Reviewer: The part with MariaDB can be omitted. This is related to DBAL and you should link to the docs of the URL like https://www.doctrine-project.org/projects/doctrine-dbal/en/2.12/reference/configuration.html#connecting-using-a-url.
Infos about MariaDB itself can be found on the same page: https://www.doctrine-project.org/projects/doctrine-dbal/en/2.12/reference/configuration.html#automatic-platform-version-detection

Reviewer 3: @SenseException there is a ton of ppl out there that were hit by this "issue" (missing or wrong serverVersion)... maybe an effective example does not hurt?

I've also noticed that for mariadb the "Automatic platform version detection" just does not work...

Reviewer: I would like to prevent a redundancy here. A link to the dbal example should be sufficient.

What about the not working automatic platform version detection? Do you want a mention that it doesn't always work and that it is a best practice to put the version into the config?

Reviewer 2: ping @goetas :slightly_smiling_face:
